ac.soton.emf.translator/src/ac/soton/emf/translator/handler/TranslateHandler.java

```diff
@@ -52,24 +50,12 @@
 	 */
 	@Override
 	public final Object execute(ExecutionEvent event) throws ExecutionException {
-		final EObject sourceElement; // = null;
+		final EObject sourceElement;
 		ISelection selection = HandlerUtil.getCurrentSelection(event);
-		if (selection instanceof TreeSelection){
-			Object obj = ((TreeSelection)selection).getFirstElement();
-			if (obj instanceof EObject){
-				sourceElement = (EObject)obj;
-			}else if (obj instanceof IInternalElement){
-				sourceElement = new EMFRodinDB().loadEventBComponent((IInternalElement)obj) ;
-			}else if (obj instanceof IAdaptable) {
-				Object adaptedObj = ((IAdaptable) obj).getAdapter(EObject.class);
-				if (adaptedObj instanceof EObject){
-					sourceElement = (EObject) adaptedObj; 
-				} else return null;
-			}else if (obj instanceof Resource){
-				sourceElement = ((Resource)obj).getContents().get(0);
-			}else return null;
-		} else return null;
-		
+		if (selection instanceof StructuredSelection && !((StructuredSelection)selection).isEmpty()){
```

Comment:
isEmpty() is defined for ISelection so no need to cast.

Reply:
Also, should use the interface IStructuredSelection rather than the implementation StructuredSelection.